lib/pbench/server/api/resources/upload_api.py

```diff
@@ -185,11 +201,13 @@ def put(self, filename: str):
                         username,
                     )
                     raise CleanupTime(
-                        HTTPStatus.INTERNAL_SERVER_ERROR, "INTERNAL ERROR"
+                        HTTPStatus.INTERNAL_SERVER_ERROR,
+                        "Duplicate dataset ID not found",
                     )
                 else:
                     response = jsonify(dict(message="Dataset already exists"))
                     response.status_code = HTTPStatus.OK
+                    self.drain(request.stream)
```

Comment:
Are you sure that you want to call `drain()` explicitly rather than trigger the cleanup actions?

Reply:
Absolutely: this is a success case, and cleanup actions would be dangerous. We do, however, need to drain the pipe to get our status through the proxy.